apps/webservice/src/utils/error/safeAwait.ts

```diff
@@ -0,0 +1,48 @@
+import type { FieldValues, UseFormReturn } from "react-hook-form";
+
+export async function safeAwait<T, E = Error>(
+  promise: Promise<T>,
+): Promise<[T, null] | [null, E]> {
+  try {
+    const result = await promise;
+    return [result, null];
+  } catch (error) {
+    return [null, error as E];
+  }
+}
```

Comment:
essenital what you have done is implemented go.

Anyway I don't think is how we should be doing this, react hooks form has error handling built into it